include/deal.II/base/mpi_noncontiguous_vector.h

```diff
@@ -0,0 +1,260 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mpi_noncontiguous_vector_h
+#  define dealii_mpi_noncontingous_vector_h
+
+#  include <deal.II/base/config.h>
+
+#  include <deal.II/base/mpi.h>
+#  include <deal.II/base/mpi_compute_index_owner_internal.h>
+
+#  include <deal.II/lac/vector_space_vector.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  namespace noncontiguous
+  {
+    template <typename Number = double>
+    class Vector
```

Comment:
Every new Vector class should derive from `VectorSpaceVector` and/or `ReadWriteVector`.

Reply:
I know ;) But before I implement all 20 methods, I wanted to ask!